### PR TITLE
Voting, Showing Results, Activating polls

### PR DESCRIPTION
**Pull request by RafalWilinski**
Closes: https://github.com/x-team/realtime-polling/issues/6, https://github.com/x-team/realtime-polling/issues/7, https://github.com/x-team/realtime-polling/issues/1, https://github.com/x-team/realtime-polling/issues/5

Video preview: https://monosnap.com/file/KFHNOQIqy0uyUMGyU4bqoaVxk5mRzO

### CODE REVIEW

**Comment by jacekelgda** [comment]
@RafalWilinski will you fix realtime in this PR or other ?

**Comment by RafalWilinski** [comment]
This one. @jacekelgda

File: src/components/Vote.vue [hunk 1]
```diff
@@ -1,26 +1,60 @@
...
+    },
   },
   methods: {
-    vote: () => {
+    vote(poll, choice) {
+      this.isVoted = true;
+
+      this.poll[choice].votes += 1;
 
+      db.ref(`/${this.$route.params.id}/${choice}`).set({
```
Comment (mmiszy) [inline src/components/Vote.vue]: What if two people vote at the same time very quickly? Isn't it possivle that some votes will be overridden?

More reliable way is to use `db.ref.transaction` like in this example: https://github.com/firebase/functions-samples/blob/3f4b76bf9d9893060b2937d47a645f1c396ec169/child-count/functions/index.js#L23-L39

File: src/components/AddPoll.vue [hunk 2]
```diff
@@ -51,6 +31,7 @@
 </template>
 <script>
 import db from '../services/firebase';
+import Results from './Results.vue';
```
Comment (mmiszy) [inline src/components/AddPoll.vue]: `Unexpected use of file extension "vue" for "./Results.vue"  `

File: src/components/Vote.vue [hunk 3]
```diff
@@ -1,26 +1,60 @@
...
+  components: {
+    Results,
+  },
   firebase: {
-    polls,
+    polls: {
+      source: polls,
+      readyCallback() {
+        this.isLoading = false;
+        this.poll = this.polls.filter(poll => poll['.key'] === this.$route.params.id)[0];
```
Comment (mmiszy) [inline src/components/Vote.vue]: `this.poll = polls.child(this.$route.params.id)`

File: src/components/Vote.vue (hunk 1 above)
Comment (RafalWilinski) [reply src/components/Vote.vue]: 🤔 Valid point. I'll fix it asap.

**Comment by RafalWilinski** [comment]
@mmiszy Code review suggestions applied.

File: src/components/Vote.vue [hunk 4]
```diff
@@ -28,25 +26,29 @@ export default {
...
+        readyCallback() {
+          this.isLoading = false;
+        },
       },
-    },
+    };
   },
   methods: {
     vote(poll, choice) {
+      const dbRef = db.ref().child(`/${this.$route.params.id}`);
```
Comment (mmiszy) [inline src/components/Vote.vue]: If I understand the docs correctly it should be possibly to simplify this logic even further by using
```
.child(`/${this.$route.params.id}/${choice}/votes`)
```

**Pull request merged**